scripts/biomedical/humanMetabolomeDatabase/format_hmdb_protein.py

```diff
@@ -0,0 +1,495 @@
+    attribute_dict = fill_gene_property(df_property, attribute_column, \
+                                            gene_attribute, "protein", "gene19", "gene38")
+    df_original[protein_dcid_column] = df_original[protein_dcid_column].\
+                                        fillna(df_original[attribute_column].map(attribute_dict))
+    df_remained = df_original[df_original[protein_dcid_column].isna()]
+
+    return df_remained
+
+
+def protein_attribute_to_protein_dcid(df_original: pd.DataFrame,
+                                      df_input: pd.DataFrame,
+                                      protein_attribute: str,
+                                      attribute_column: str,
+                                      protein_dcid_column: str) -> pd.DataFrame:
+    """Filling protein_dcid_column of a dataframe
+    using a protein attribute column and datacommons
+    Args:
+        df_original: dataframe with protein dcid column to be imputed
+        df: dataframe where attribute column is extracted
+        protein_attribute: datacommons property name of protein
+        attribute_col: attribute column
+        protein_dcid_column: protein dcid column to be imputed
+    Returns:
+        df_remained: dataframe with protein dcid missing
+    """
+    all_results = []
+    attribute = list(set(df_input[attribute_column].unique()) - set([np.nan]))
+    for alist in split_list(attribute, QUERY_SIZE):
+        attribute = np.array(alist)
+        query_str = """
+        SELECT DISTINCT ?protein
+        WHERE {{
+        ?protein typeOf Protein .
+        ?protein {0} {1} .
+        }}
+        """.format(protein_attribute, attribute)
+        result = dc.query(query_str)
+        all_results.extend(result)
+    protein_dcid = np.array([res["?protein"] for res in all_results])
+    df_property = pd.DataFrame(columns=[attribute_column, "protein"])
+    df_property["protein"] = protein_dcid
+
+    property_dict = dc.get_property_values(protein_dcid,
+                                           prop=protein_attribute,
+                                           out=True)
+    df_property[attribute_column] = df_property["protein"].map(property_dict)
+    df_property = df_property.explode(attribute_column)
+    property_dict = df_property.set_index(attribute_column).to_dict()["protein"]
+    df_original[protein_dcid_column] = df_original[protein_dcid_column].\
+                                fillna(df_original[attribute_column].map(property_dict))
+    df_remained = df_original[df_original[protein_dcid_column].isna()]
+    return df_remained
+
+
+def format_comparment(compartment: str) -> str:
+    """format subcellular_location naming
+    to datacommons format: drop bad values,
+    synchronize synonyms, Enum format
+    Args:
+        compartment: compartment name
+    Returns:
+        compartment: formatted compartment
+    """
+    modify_dict = {"Cytoplasmic": "Cytoplasm", "Cytoplasmic side": "Cytoplasm"}

```

Comment:
usually for constants in python, we want to name the variable in all caps

Reply:
@khoahoang1891999